query-engine/core/src/query_graph/formatters.rs

```diff
@@ -0,0 +1,123 @@
+use super::*;
+use std::fmt::{self, Display};
```

Comment:
See if everything here can implement `Display`.

Reply:
Using the `Formatter` is a better idea than allocating a new `String` every time.